### ISSUE TITLE
SEQR local_install Docker_compose for config.yaml missing 

### ISSUE THREAD

**Issue by samarthaFFM** [macarthur-lab/seqr#1248]
https://github.com/macarthur-lab/seqr/blob/master/deploy/LOCAL_INSTALL.md

**Comment by samarthaFFM** [comment]
Hi guys,

I am trying the new local on-prem docker install of Seqr and I am trying to wget docker-compose.yml from the hail_v02_pipeline step, but it does not exist and causes an Error 404. Also, I tried the  docker-compose.yml (https://github.com/macarthur-lab/seqr/blob/dev/docker-compose.yml/) and it resulted in an error as well -  ERROR: yaml.scanner.ScannerError: mapping values are not allowed here
  in "./docker-compose.yml", line 115, column 73

Could you please let me know how to install it? thanks,

**Comment by samarthaFFM** [comment]
Also, a way to split the on-prem commands for annotating the vcf file on multiple machines and loading it on a new elastisearch server would be nice as well. Thanks

**Comment by bw2** [comment]
Hi @samarthaFFM , we need to update the link to 
https://raw.githubusercontent.com/macarthur-lab/seqr/master/docker-compose.yml
Please wget that link instead.

The  ERROR: yaml.scanner.ScannerError was because 
https://github.com/macarthur-lab/seqr/blob/dev/docker-compose.yml is not the raw version of the file and contains html.

If you're interested in annotating a vcf using multiple machines on-prem, you will need to have on-prem Spark cluster installed across multiple machines, and submit your annotation job to that.

**Comment by samarthaFFM** [comment]
Hi @bw2 , thank you for the quick reply. Is it also possible to split the jobs into multiple elastisearch or Seqr instances as well? Currently I see that you can provide a different elastisearch host to the python loading script. But I was wondering if the same is possible of a SEQR instance and to copy the elastisearch indices from one Seqr instance to the other? Thanks.

**Comment by samarthaFFM** [comment]
The problem of our on-prem installation is lack of sufficient resources , so we are planning to run it on multiple on-prem servers.

**Comment by bw2** [comment]
docker-compose only supports running components on a single machine, but there might be networking strategies to run several docker-compose instances on different machines. I don't have experience with this though. If you'd like to split seqr across mutltiple machines, II would recommend using docker-compose for just the seqr/phenotips/redis/postgres components and [manually install](https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html#_installing_elasticsearch_yourself) multiple inter-connected elasticsearch instances on separate machines outside of docker-compose. The most difficult part would probably be to set up the on-prem spark cluster for running hail pipelines, unless you have this already.

**Comment by samarthaFFM** [comment]
Hi @bw2 , thank you so much for the input, I will get on installing the spark cluster. I tried doing the SEQR install, but the Phenotips container is unhealthy all the time. 

[root@vls191-seqr Seqr_install]# docker-compose up -d seqr
Starting seqr_install_elasticsearch_1 ...
Starting seqr_install_elasticsearch_1 ... done
Starting seqr_install_postgres_1      ... done

ERROR: for phenotips  Container "ba1237604f42" is unhealthy.
ERROR: Encountered errors while bringing up the project.

Is there any fix to this? Thanks

**Comment by bw2** [comment]
Hi @samarthaFFM 

This should be working. 
Can you try deleting the `data` subdirectory in the dir. where you ran docker-compose, and then running  `docker-compose up seqr` again?

If it's still not working, can you run `docker-compose logs phenotips`?
The logs should end with:
```
phenotips_1        | WARNING: Line 109754: Expected that dbObject "81" maps to symbol "DECIPHER:81" but it maps to "ORPHANET:81"
phenotips_1        | Apr 23, 2020 4:31:35 AM ontologizer.association.AssociationParser$1GAFByteLineScanner newLine
phenotips_1        | WARNING: Line 109755: Expected that dbObject "81" maps to symbol "DECIPHER:81" but it maps to "ORPHANET:81"
phenotips_1        | Apr 23, 2020 4:31:35 AM ontologizer.association.AssociationParser$1GAFByteLineScanner newLine
phenotips_1        | WARNING: Line 109756: Expected that dbObject "81" maps to symbol "DECIPHER:81" but it maps to "ORPHANET:81"
phenotips_1        | Apr 23, 2020 4:31:35 AM ontologizer.association.AssociationParser$1GAFByteLineScanner newLine
phenotips_1        | WARNING: Line 109757: Expected that dbObject "81" maps to symbol "DECIPHER:81" but it maps to "ORPHANET:81"
phenotips_1        | Apr 23, 2020 4:31:35 AM ontologizer.association.AssociationParser importAssociationFile
phenotips_1        | INFO: 112327 associations parsed, 112327 of which were kept while 0 malformed lines had to be ignored.
phenotips_1        | Apr 23, 2020 4:31:35 AM ontologizer.association.AssociationParser importAssociationFile
phenotips_1        | INFO: A further 0 associations were skipped due to various reasons whereas 0 of those where explicitly qualified with NOT, 0 referred to obsolete terms and 0 didn't match the requested evidence codes
phenotips_1        | Apr 23, 2020 4:31:35 AM ontologizer.association.AssociationParser importAssociationFile
phenotips_1        | INFO: A total of 6688 terms are directly associated to 7519 items.
phenotips_1        | Apr 23, 2020 4:31:36 AM ontologizer.go.Ontology assignLevel1TermsAndFixRoot
phenotips_1        | INFO: Ontology contains a single level-one term (All (HP:0000001)
phenotips_1        | 2020-04-23 04:31:38.494:INFO:optjl.NotifyListener:Server started, you can now open http://8f31c183148e:8080/ in your browser to access PhenoTips.
phenotips_1        | 2020-04-23 04:31:38.495:INFO:optjl.NotifyListener:----------------------------------
phenotips_1        | 2020-04-23 04:31:38.527:WARN:optjl.OpenBrowserListener:Desktop API is not available.
phenotips_1        | 2020-04-23 04:31:40,669 [XWiki initialization] INFO  .HibernateDataMigrationManager - Storage schema updates and data migrations are enabled
phenotips_1        | 2020-04-23 04:31:40,900 [XWiki initialization] INFO  .HibernateDataMigrationManager - No data migration to apply for wiki [xwiki] currently in version [54693]
phenotips_1        | 2020-04-23 04:31:40,903 [XWiki initialization] INFO  .HibernateDataMigrationManager - Checking Hibernate mapping and updating schema if needed for wiki [xwiki]
```

If you see something else, please share your phenotips logs.

**Comment by samarthaFFM** [comment]
Hi @bw2 , thank you so much for the input, I tried to delete the Data folder and docker-compose Seqr again. It was the same issue. The phenotips log created a huge unending file. I have put a few lines from the log here - 

Attaching to seqr_install_phenotips_1
ESC[36mphenotips_1        |ESC[0m PHENOTIPS_SERVICE_PORT=8080
ESC[36mphenotips_1        |ESC[0m HOSTNAME=501d02cdb6fc
ESC[36mphenotips_1        |ESC[0m PT_VERSION=1.2.6
ESC[36mphenotips_1        |ESC[0m PWD=/phenotips-standalone-1.2.6
ESC[36mphenotips_1        |ESC[0m HOME=/root
ESC[36mphenotips_1        |ESC[0m SHLVL=1
ESC[36mphenotips_1        |ESC[0m PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
ESC[36mphenotips_1        |ESC[0m _=/usr/bin/env
ESC[36mphenotips_1        |ESC[0m Current Directory: /phenotips-standalone-1.2.6
ESC[36mphenotips_1        |ESC[0m + psql --host postgres -U postgres -l
ESC[36mphenotips_1        |ESC[0m + grep xwiki
ESC[36mphenotips_1        |ESC[0m psql: could not translate host name "postgres" to address: Temporary failure in name resolution
ESC[36mphenotips_1        |ESC[0m + psql --host postgres -U postgres -c 'CREATE DATABASE xwiki'
ESC[36mphenotips_1        |ESC[0m psql: could not translate host name "postgres" to address: Temporary failure in name resolution
ESC[36mphenotips_1        |ESC[0m + psql --host postgres -U postgres -c 'CREATE ROLE xwiki WITH CREATEDB LOGIN PASSWORD '\''xwiki'\'''
ESC[36mphenotips_1        |ESC[0m psql: could not translate host name "postgres" to address: Temporary failure in name resolution
ESC[36mphenotips_1        |ESC[0m + psql --host postgres -U postgres -c 'GRANT ALL PRIVILEGES ON DATABASE xwiki TO xwiki'
ESC[36mphenotips_1        |ESC[0m psql: could not connect to server: Connection refused
ESC[36mphenotips_1        |ESC[0m       Is the server running on host "postgres" (172.18.0.3) and accepting
ESC[36mphenotips_1        |ESC[0m       TCP/IP connections on port 5432?
ESC[36mphenotips_1        |ESC[0m + PGPASSWORD=xwiki
ESC[36mphenotips_1        |ESC[0m + psql --host postgres -U xwiki xwiki -f /init_phenotips_db.sql
ESC[36mphenotips_1        |ESC[0m psql: could not translate host name "postgres" to address: Temporary failure in name resolution
ESC[36mphenotips_1        |ESC[0m + ./start.sh
ESC[36mphenotips_1        |ESC[0m Starting Jetty on port 8080, please wait...
ESC[36mphenotips_1        |ESC[0m OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=192m; support was removed in 8.0
ESC[36mphenotips_1        |ESC[0m 2020-04-17 16:44:18,617 [main] WARN  o.a.s.c.SolrResourceLoader     - Can't find (or read) directory to add to classloader: lib (resolved as: /phenotips-standalone-1.2.6/data/solr/lib).
ESC[36mphenotips_1        |ESC[0m 2020-04-17 16:44:21,500 [coreLoadExecutor-5-thread-1] WARN  o.a.s.c.SolrCore               - [chebi] Solr index directory '/phenotips-standalone-1.2.6/data/solr/chebi/data/index' doesn't exist. Creating new index...
ESC[36mphenotips_1        |ESC[0m 2020-04-17 16:44:23,464 [coreLoadExecutor-5-thread-3] WARN  o.a.s.c.SolrCore               - [patients] Solr index directory '/phenotips-standalone-1.2.6/data/solr/patients/data/index' doesn't exist. Creating new index...
ESC[36mphenotips_1        |ESC[0m Apr 17, 2020 4:44:26 PM ontologizer.FileCache setCacheDirectory
ESC[36mphenotips_1        |ESC[0m INFO: Cache directory set to "/phenotips-standalone-1.2.6/jetty/work/jetty-0.0.0.0-8080-phenotips-_-any-/xwiki-temp/ontologizer/.cache"
ESC[36mphenotips_1        |ESC[0m Apr 17, 2020 4:44:26 PM ontologizer.FileCache setCacheDirectory
ESC[36mphenotips_1        |ESC[0m WARNING:
ESC[36mphenotips_1        |ESC[0m java.io.FileNotFoundException: /phenotips-standalone-1.2.6/jetty/work/jetty-0.0.0.0-8080-phenotips-_-any-/xwiki-temp/ontologizer/.cache/.index (No such file or directory)
ESC[36mphenotips_1        |ESC[0m       at java.io.FileInputStream.open0(Native Method)
ESC[36mphenotips_1        |ESC[0m       at java.io.FileInputStream.open(FileInputStream.java:195)
ESC[36mphenotips_1        |ESC[0m       at java.io.FileInputStream.<init>(FileInputStream.java:138)
ESC[36mphenotips_1        |ESC[0m       at java.io.FileReader.<init>(FileReader.java:72)
ESC[36mphenotips_1        |ESC[0m       at ontologizer.FileCache.setCacheDirectory(FileCache.java:302)
ESC[36mphenotips_1        |ESC[0m       at org.phenotips.diagnosis.internal.BoqaUtils.loadDataFiles(BoqaUtils.java:64)
ESC[36mphenotips_1        |ESC[0m       at org.phenotips.diagnosis.internal.DefaultDiagnosisService.initialize(DefaultDiagnosisService.java:109)
ESC[36mphenotips_1        |ESC[0m       at org.xwiki.component.embed.InitializableLifecycleHandler.handle(InitializableLifecycleHandler.java:39)
ESC[36mphenotips_1        |ESC[0m       at org.xwiki.component.embed.EmbeddableComponentManager.createInstance(EmbeddableComponentManager.java:322)
ESC[36mphenotips_1        |ESC[0m       at org.xwiki.component.embed.EmbeddableComponentManager.getComponentInstance(EmbeddableComponentManager.java:424)
ESC[36mphenotips_1        |ESC[0m       at org.xwiki.component.embed.EmbeddableComponentManager.getComponentInstance(EmbeddableComponentManager.java:392)
ESC[36mphenotips_1        |ESC[0m       at org.xwiki.component.embed.EmbeddableComponentManager.getInstance(EmbeddableComponentManager.java:190)
ESC[36mphenotips_1        |ESC[0m       at org.xwiki.component.embed.EmbeddableComponentManager.getDependencyInstance(EmbeddableComponentManager.java:363)

**Comment by samarthaFFM** [comment]
Hi @bw2 

Additionally, from the past few days, I have been having issues in uploading the data to the elastisearch host. 

java.io.IOException: Error listing: bucket: seqr-reference-data, object: GRCh37/all_reference_data/combined_reference_data_grch37.vds/rdd.parquet/

Is this something related to downloading data from somewhere not local? Thanks in advance
